src/Interfaces/AlgorithmInterface.php

```diff
@@ -0,0 +1,8 @@
+<?php
+
+namespace Kmeans\Interfaces;
+
+interface AlgorithmInterface
+{
+    public function clusterize(PointCollectionInterface $points, int $nbClusters): ClusterCollectionInterface;
```

Comment:
https://github.com/bdelespierre/php-kmeans/issues/32

In that discussion we decided to add setDistanceFunction to AlgorithmInterface .

Reply:
Yes, in another PR :+1: